### PR TITLE
Feat 38 codeocean results

### LINKED ISSUE #38: Indexer should include processed Code Ocean results
The indexer needs to include assets that are processed results in the Code Ocean datasets bucket. This is needed so that science teams can analyze data as soon as it is processed, regardless of whether we are capturing it to an external bucket.

**Acceptance Criteria**
- [ ] The indexer crawls the Code Ocean datasets bucket.
- [ ] The indexer looks at all top-level prefixes. If the prefix contains (*not* recursively) any of the well-known JSON files (data_description.json, subject.json, procedures.json, rig.json, session.json, instrument.json, acquisition.json), include it in the index.
- [ ] Query Code Ocean to populate relevant Metadata record fields (name, creation time, etc).

### PR DESCRIPTION
**Pull request by jtyoung84**
Closes #38

 - Adds job to index processed results hosted in Code Ocean bucket
 - Compares `locations` from Code Ocean index to `locations` in DocDB
   - Adds new items to DocDB
   - Deletes records from DocDB if location isn't found in Code Ocean response

### CODE REVIEW

File: src/aind_data_asset_indexer/models.py [hunk 1]
```diff
@@ -109,3 +109,64 @@ class AindIndexBucketsJobSettings(AindIndexBucketJobSettings):
...
+        param_store_name : str
+        """
+        param_store_client = boto3.client("ssm")
+        response = param_store_client.get_parameter(
+            Name=param_store_name, WithDecryption=True
+        )
+        param_store_client.close()
+        parameters = response["Parameter"]["Value"]
+        parameters_json = json.loads(parameters)
+        if "doc_db_secret_name" not in parameters:
```
Comment (helen-m-lin) [inline src/aind_data_asset_indexer/models.py:141]: nit: I was checking values from the `parameters` string and not the `parameters_json` keys in `AindIndexBucketJobSettings`. It works as-is but is not ideal. Let me know if you want me to fix this in both models.

```suggestion
        if "doc_db_secret_name" not in parameters_json:
```

File: src/aind_data_asset_indexer/models.py [hunk 2]
```diff
@@ -109,3 +109,64 @@ class AindIndexBucketsJobSettings(AindIndexBucketJobSettings):
...
+            raise ValueError("doc_db_secret_name not found in parameters.")
+        if "codeocean_secret_name" not in parameters:
```
Comment (helen-m-lin) [inline src/aind_data_asset_indexer/models.py:143]:
```suggestion
        if "codeocean_secret_name" not in parameters_json:
```

**Review by helen-m-lin — APPROVED** [review]
Everything looks great. We will have to update the parameters and some permissions in the ecs task later and update `run.sh`.

File: src/aind_data_asset_indexer/models.py (hunk 1 above)
Comment (jtyoung84) [reply src/aind_data_asset_indexer/models.py:141]: I'll fix it. I think you needed to check `parameters_json.keys()`

Comment (helen-m-lin) [reply src/aind_data_asset_indexer/models.py]: Thanks!

**Pull request merged**